src/sgp/SoundMan.cc

```diff
@@ -723,11 +723,13 @@ static SOUNDTAG* SoundGetChannelByID(UINT32 uiSoundID)
 
 static void SoundCallback(void* userdata, Uint8* stream, int len)
 {
-	SDL_memset(stream, 0, len);
+	INT32  mix[ gTargetAudioSpec.samples ];
 
-	UINT16* Stream = (UINT16*)stream;
+	Assert(len * 2 == sizeof(mix));
```

Comment:
Is there documentation somewhere that this is always true?

Reply:
Your result below proves that it's not true.
Looks like SDL can change the number of samples it asks for in the callback.